18F/identity-idp
# LG-9321: New field on Enrollment Outcomes for GetUspsProofingResultsJob Summary

## 🎫 Ticket
[LG-9321](https://cm-jira.usa.gov/browse/LG-9321)

## 🛠 Summary of changes

- Calculate percentage of enrollments errored when enrollments exists
- Add key/value to enrollment_outcomes,key=percent_enrollments_errored (to be consumed in identity-devops by my new HighErrorRate alarm, PR coming soon)
- Update and add testing

## 📜 Testing Plan

- [ ] Observer new alarm IDPUSPSProofingWorker-HighErrorRate in [Slack](https://gsa-tts.slack.com/archives/C043ZSQ3NUV/p1681491811238769). This shows the new key/value was consumed.

- [ ] Observe newly created field and calculated value in AWS CloudWatch.

1. Sign in to AWS with `aws-vault login sandbox-power`
2. Go to CloudWatch
3. Go to Logs Insights
4. Pick `joy_/srv/idp/shared/log/events.log` as the log group
5. Update the query to be...
`fields @timestamp, @message, @logStream, @log | filter name like "GetUspsProofingResultsJob: Job completed"`
6. Select Custom (top right corner) and pick April 14, 2023 and then click Run query.
7. Open up the logs. All but the first will have `properties.event_properties.percent_enrollments_errored` listed with the correctly calculated percentage,  (err/checked)*100. 

- [ ] Updated x1 test, added 2 additional tests

## 👀 Screenshots

<details>
<summary>Before:</summary>

Field _properties.event_properties.percent_enrollments_errored_  **is not** on summary.

<img width="625" alt="Screenshot 2023-04-14 at 1 07 43 PM" src="https://user-images.githubusercontent.com/125507397/232134782-415c8e2e-96d0-4f65-b511-e4aad896fcee.png">
</details>

<details>
<summary>After:</summary>

Field _properties.event_properties.percent_enrollments_errored_  **is** on summary with the correctly calculated percentage.


<img width="682" alt="Screenshot 2023-04-14 at 1 07 27 PM" src="https://user-images.githubusercontent.com/125507397/232134890-0b82b2af-0c75-4a27-b270-f631589c1708.png">

<img width="702" alt="Screenshot 2023-04-14 at 1 17 54 PM" src="https://user-images.githubusercontent.com/125507397/232136539-1d69c201-2878-4600-88d6-1a5dd328e496.png">


</details>
